airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/ConnectionManagerWorkflow.java

```diff
@@ -53,8 +56,26 @@ public interface ConnectionManagerWorkflow {
   @SignalMethod
   void connectionUpdated();
 
+  @Data
+  @NoArgsConstructor
+  @AllArgsConstructor
+  @Builder
+  class ResetInput {
+
+    private boolean isGlobal;
+    private Set<String> streamNames;
```

Comment:
A couple questions about this:
1. Is the idea here that if `isGlobal` is set to `true`, then `streamNames` is just ignored completely? It would be nice if this was enforced using Java types but there may not be any great way to do that.
2. Will we for sure know what the value of `isGlobal` should be at the time that this input is constructed?
3. Is it potentially redundant to have both of these properties in the input? I.e. instead we could have only `streamNames`, and in the global case we just pass in all stream names for the connection? Though maybe this has the potential to get out of sync with the full set of streams if a new one is added right after this input is generated, so this may not be the best idea.

Reply:
1. Yes, it is the main goal is to make temporal to fetch the stream to reset at the moment of when the reset is run. 
2. I have updated with defaultBuilder value and a test
3. My idea was to make sure that we fetch is as close as the creation of the job config in the workflow. So make this fetch go through an activity. I wonder if the can be a race condition like: send reset ->  add stream x y z -> temporal start running the asynchronous signal method. If we pass the stream through the input we will miss x y z.

Reply:
Got it, your response to point 3 makes sense. If in the `isGlobal: true` case we only fetch the full list of streams to reset right before we create the job, then we should be guaranteed that no new streams will be added to the destination between that time and when the job is ran, because this connection manager workflow should be the only one running for this connection.